### ISSUE TITLE
Newrelic-daemon keeps getting stopped by Puppet and then restarted by PHP-FPM.

### ISSUE THREAD

**Issue by Yggdrasil** [claranet/puppet-newrelic#13]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 4.10.9
- Distribution: Ubuntu 14.04
- Module version: 2.01

## How to reproduce (e.g Puppet code you use)

Using PHP-FPM 7.1 packages from Ondrej's PPA (managed via module puppet-php v5.1.0).

Hiera:
```
newrelic::license_key: 1235467890etc...
newrelic::enable_infra: false
newrelic::enable_server: true
newrelic::enable_php_agent: true
newrelic::agent::php::conf_dir: /etc/php/7.1/mods-available
newrelic::agent::php::purge_files: [ '/etc/php/7.1/apache2/conf.d/newrelic.ini','/etc/php/7.1/fpm/conf.d/newrelic.ini' ]
newrelic::agent::php::ini_settings: { 'appname': "%{::hostname} PHP" }
```

roles::lamp_server
```
class roles::lamp_server {
  contain profiles::newrelic
  contain profiles::apache
  contain profiles::php
  ...
}
```

profiles::newrelic:
```
class profiles::newrelic {
  contain newrelic
}
```

## What are you seeing
I'm using the module to deploy both newrelic-sysmond and the PHP agent. The newrelic-daemon process is automatically started by PHP-FPM. If the agent `startup_mode` is set to 'agent' (this is the default in `newrelic::agent::php`) the module stops the `newrelic-daemon` process on each Puppet run. It is then started again by PHP-FPM, and stopped again by Puppet, etc., etc...

The daemon's flip-flopping leads to continuous changes in the Puppet logs and dashboards.

## What behaviour did you expect instead
I think the automatic starting of `newrelic-daemon` by the PHP extension is meant to happen. It appears this module doesn't take that into account.

I expect the module to install and configure the agent's settings and stop the `newrelic-daemon` service from being managed by the OS's init scripts. It should then be left alone to be managed by the PHP extension.

## Output log
Puppet debug log from a node:
```
Debug: Exec[install_newrelic_license_key](provider=posix): Executing check '/bin/grep -q cfd5f83246c2c117d213eeddf4087e3f452d95b1 /etc/newrelic/nrsysmond.cfg'
Debug: Executing with uid=root: '/bin/grep -q cfd5f83246c2c117d213eeddf4087e3f452d95b1 /etc/newrelic/nrsysmond.cfg'
Debug: Service[newrelic-sysmond](provider=upstart): Could not find newrelic-sysmond.conf in /etc/init
Debug: Service[newrelic-sysmond](provider=upstart): Could not find newrelic-sysmond.conf in /etc/init.d
Debug: Service[newrelic-sysmond](provider=upstart): Could not find newrelic-sysmond in /etc/init
Debug: Executing: '/usr/sbin/service newrelic-sysmond status'
Debug: Exec[newrelic install](provider=posix): Executing check '/bin/grep -q cfd5f83246c2c117d213eeddf4087e3f452d95b1 /etc/php/7.1/mods-available/newrelic.ini'
Debug: Executing with uid=root: '/bin/grep -q cfd5f83246c2c117d213eeddf4087e3f452d95b1 /etc/php/7.1/mods-available/newrelic.ini'
Debug: /Stage[main]/Newrelic::Agent::Php/File[/etc/php/7.1/apache2/conf.d/newrelic.ini]: Nothing to manage: no ensure and the resource doesn't exist
Debug: /Stage[main]/Newrelic::Agent::Php/File[/etc/php/7.1/fpm/conf.d/newrelic.ini]: Nothing to manage: no ensure and the resource doesn't exist
Debug: /Stage[main]/Newrelic::Agent::Php/File[/etc/newrelic/newrelic.cfg]: Nothing to manage: no ensure and the resource doesn't exist
Debug: Service[newrelic-daemon](provider=upstart): Could not find newrelic-daemon.conf in /etc/init
Debug: Service[newrelic-daemon](provider=upstart): Could not find newrelic-daemon.conf in /etc/init.d
Debug: Service[newrelic-daemon](provider=upstart): Could not find newrelic-daemon in /etc/init
Debug: Executing: '/usr/sbin/service newrelic-daemon status'
Debug: Executing: '/etc/init.d/newrelic-daemon stop'
Notice: /Stage[main]/Newrelic::Agent::Php/Service[newrelic-daemon]/ensure: ensure changed 'running' to 'stopped'
Debug: /Stage[main]/Newrelic::Agent::Php/Service[newrelic-daemon]: The container Class[Newrelic::Agent::Php] will propagate my refresh event
Debug: Class[Newrelic::Agent::Php]: The container Stage[main] will propagate my refresh event
```

## Any additional information you'd like to impart
I'm not too familiar with the module or the Agent so perhaps I can configure it a different way to prevent this behaviour, but I don't see it.

**Comment by craigwatson** [comment]
Hi,

Thanks for raising this, it is indeed a module bug as I was able to reproduce it using the module's PHP-FPM test-case, using the Slashbunny PHP-FPM module rather than Voxpupuli.

The bug has been fixed in 2f0c6d39fb9830e0ad73df43a157eb938a4d4cf5 and version 2.1.0 has been released to the Forge.

If you still see the issue after upgrading, please let us know and I'll look into it further.

Thanks!
Craig

**Comment by Yggdrasil** [comment]
Thanks for the excellent response @craigwatson. Version 2.1.0 does indeed work as expected.
